api/notd/api/endpoints_v1.py

```diff
@@ -9,15 +9,41 @@
 from notd.api.models_v1 import ApiCollection
 from notd.api.models_v1 import ApiCollectionToken
 from notd.api.models_v1 import ApiTokenTransfer
-from notd.api.models_v1 import ApiUiData
 
 
-class RetrieveUiDataRequest(BaseModel):
+class  RetrievedHighestPriceTransferRequest(BaseModel):
     startDate: Optional[datetime.datetime]
     endDate: Optional[datetime.datetime]
 
-class RetrieveUiDataResponse(BaseModel):
-    uiData: ApiUiData
+class RetrievedHighestPriceTransferResponse(BaseModel):
+    transfer: ApiTokenTransfer
+
+class  RetrievedRandomTransferRequest(BaseModel):
+    startDate: Optional[datetime.datetime]
+    endDate: Optional[datetime.datetime]
+
+class RetrievedRandomTransferResponse(BaseModel):
+    transfer: ApiTokenTransfer
+
+class  RetrievedTransactionCountRequest(BaseModel):
+    startDate: Optional[datetime.datetime]
+    endDate: Optional[datetime.datetime]
+
+class RetrievedTransactionCountResponse(BaseModel):
+    count: int
+
+class  RetrievedMostTradedRequest(BaseModel):
+    startDate: Optional[datetime.datetime]
+    endDate: Optional[datetime.datetime]
+
+class RetrievedMostTradedResponse(BaseModel):
+    tokenTransfers: List[ApiTokenTransfer]
```

Comment:
this one is really inefficient, we should create a new object with:
- collectionToken: ApiCollectionToken
- latestTransfer: ApiTokenTransfer
- transferCount: int